spring-kafka/src/test/java/org/springframework/kafka/listener/ShareKafkaMessageListenerContainerIntegrationTests.java

```diff
@@ -96,28 +111,562 @@ void integrationTestShareKafkaMessageListenerContainer(EmbeddedKafkaBroker broke
+
+			// Wait for redelivery of released record
+			assertThat(redeliveryLatch.await(15, TimeUnit.SECONDS)).isTrue();
+			assertThat(ackTypes.get("release")).isEqualTo(AcknowledgeType.ACCEPT); // what got released, was accepted eventually.
+			assertThat(ackTypes.get("release-redelivered")).isEqualTo(AcknowledgeType.RELEASE);
+
+		}
+		finally {
+			container.stop();
+		}
+	}
+
+	@Test
+	void shouldSupportDifferentListenerTypes(EmbeddedKafkaBroker broker) throws Exception {
+		String topic = "share-container-implicit-test";
+		String groupId = "share-container-listener-types-group";
+		String bootstrapServers = broker.getBrokersAsString();
+
+		setShareAutoOffsetResetEarliest(bootstrapServers, groupId);
+		produceTestRecords(bootstrapServers, topic, 1);
+
+		Map<String, Object> consumerProps = createConsumerProps(bootstrapServers, groupId, false);
+		DefaultShareConsumerFactory<String, String> factory = new DefaultShareConsumerFactory<>(consumerProps);
+
+		// Test 1: Basic MessageListener
+		testBasicMessageListener(factory, topic, bootstrapServers, groupId + "-basic");
+
+		// Test 2: ShareConsumerAwareMessageListener
+		testShareConsumerAwareListener(factory, topic, bootstrapServers, groupId + "-aware");
+
+		// Test 3: AcknowledgingShareConsumerAwareMessageListener in implicit mode
+		testAckListenerInImplicitMode(factory, topic, bootstrapServers, groupId + "-ack-implicit");
+	}
+
+	@Test
+	void shouldHandleContainerLifecycle(EmbeddedKafkaBroker broker) throws Exception {
+		String topic = "share-container-lifecycle-test";
+		String groupId = "share-container-lifecycle-group";
+		String bootstrapServers = broker.getBrokersAsString();
+
+		setShareAutoOffsetResetEarliest(bootstrapServers, groupId);
+
+		Map<String, Object> consumerProps = createConsumerProps(bootstrapServers, groupId, true);
+		DefaultShareConsumerFactory<String, String> factory = new DefaultShareConsumerFactory<>(consumerProps);
+
+		ContainerProperties containerProps = new ContainerProperties(topic);
+		containerProps.setExplicitShareAcknowledgment(true);
+
+		AtomicBoolean listenerCalled = new AtomicBoolean(false);
+		containerProps.setMessageListener(new AcknowledgingShareConsumerAwareMessageListener<String, String>() {
+			@Override
+			public void onShareRecord(ConsumerRecord<String, String> record,
+					@Nullable ShareAcknowledgment acknowledgment, ShareConsumer<?, ?> consumer) {
+				listenerCalled.set(true);
+				acknowledgment.acknowledge();
+			}
+		});
+
+		ShareKafkaMessageListenerContainer<String, String> container =
+				new ShareKafkaMessageListenerContainer<>(factory, containerProps);
+		container.setBeanName("lifecycleTestContainer");
+
+		// Test initial state
+		assertThat(container.isRunning()).isFalse();
+
+		// Test start
+		container.start();
+		assertThat(container.isRunning()).isTrue();
+
+		// Test processing
+		produceTestRecords(bootstrapServers, topic, 1);
+		Thread.sleep(3000); // Give time for processing
```

Comment:
Cannot we use a `CountDownLatch` instead?
Might be the case that even these 3 seconds are not enough.
Or opposite: the processing is so fast that we block here for nothing.